src/crimson/osd/main.cc

```diff
@@ -21,33 +21,66 @@ void usage(const char* prog) {
   generic_server_usage();
 }
 
+auto partition_args(seastar::app_template& app, int argc, char* argv[])
+{
+  namespace bpo = boost::program_options;
+  // collect all options consumed by seastar::app_template
+  auto parsed = bpo::command_line_parser(argc, argv)
+    .options(app.get_options_description()).allow_unregistered().run();
+  const auto unknown_args = bpo::collect_unrecognized(parsed.options,
+                                                      bpo::include_positional);
+  std::vector<const char*> ceph_args, app_args;
+  bool consume_next_arg = false;
+  std::partition_copy(
+    argv, argv + argc,
+    std::back_inserter(ceph_args),
+    std::back_inserter(app_args),
+    [begin=unknown_args.begin(),
+     end=unknown_args.end(),
+     &consume_next_arg](char* arg) {
+      if (std::find(begin, end, arg) != end) {
+        return true;
+        // ceph_argparse_early_args() and
```

Comment:
Wouldn't be more readable to have this comment under the `else if (std::strcmp(arg, "-c") == 0) { ... }` block?

Reply:
fixed and repushed.